src/main/java/Decide.java

```diff
@@ -43,4 +43,64 @@ public static boolean LIC0(Point[] pts, Parameters params) {
         }
         return false;
     }
-}
\ No newline at end of file
+    /**
+     * Checks if there exists at least one set of n_pts consecutive data points such that
+     * at least one of the points lies a distance greater than dist from the line joining
+     * the first and last of these n_pts points.
+     * @param pts array of Point objects
+     * @param params Parameter object
+     * @return true if the distance between line and point is greater than params.dist.
+     */
+    public static boolean LIC6(Point[] pts, Parameters params) {
+        double dxLine, dyLine;
+        double temp, u;
+        double dxPoint, dyPoint;
+        double x, y;
+        double lineDist;
+
+        if (params.dist < 0)
+            throw new IllegalArgumentException("dist must be greater than or equal to 0");
+
+        if (params.n_pts < 3 || pts.length < 3 || pts.length < params.n_pts) {
+            return false;
+        }
+        for (int i = 0; i <= pts.length-params.n_pts; i++) {
+            for (int j = 1; j < params.n_pts; j++) {
+
+                // If the first and last points of the n_pts are identical, the distance to
+                // compare with dist will be the distance from the coincident point to all
+                // other points of the n_pts consecutive points.
+                if ((pts[i].X == pts[i+params.n_pts-1].X) &&
+                        pts[i].Y == pts[i+params.n_pts-1].Y) {
+                    return (pts[i].distance(pts[i+j]) > params.dist);
```

Comment:
You have to check all of the n_pts points and see if they meet the condition. Here you only check the point directly after point i. So if this point does not fulfill the requirement your code will return false, when it really should check the other points. You should instead have: 
if(pts[i].distance(pts[i+j]) > params.dist) return true

Reply:
You're right, a fix has been pushed!